java/tests/src/test/java/org/signal/libsignal/crypto/CryptographicHashTests.java

```diff
@@ -0,0 +1,57 @@
+//
+// Copyright 2021 Signal Messenger, LLC.
+// SPDX-License-Identifier: AGPL-3.0-only
+//
+
+package org.signal.libsignal.crypto;
+import junit.framework.TestCase;
+import org.signal.libsignal.crypto.CryptographicHash;
+import org.whispersystems.libsignal.util.Hex;
+
+public class CryptographicHashTests extends TestCase {
+
+   void hashKat(String algo, String hexInput, String hexExpectedOutput) throws Exception {
+     CryptographicHash hash = new CryptographicHash(algo);
+
+     byte[] input = Hex.fromStringCondensed(hexInput);
+
+     hash.update(input);
+     byte[] digestAllInOne = hash.finish();
+
+     assertEquals(Hex.toHexString(digestAllInOne), hexExpectedOutput);
+
+     if(input.length > 1) {
```

Comment:
Why this check?

Reply:
I guess not strictly necessary, but without it the test function would fail for an empty input.

Reply:
:facepalm: Right, my brain saw the condition and flagged it even after I'd looked at what was being tested. My bad.